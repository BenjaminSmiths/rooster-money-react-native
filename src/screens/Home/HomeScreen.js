import { observer } from 'mobx-react'
import React from 'react'
import { Component } from 'react'
import { StyleSheet, Dimensions, Image, View, Text } from 'react-native'
import styled from 'styled-components/native'
import backgroundBlue from '../../images/background-blue.jpg'
import parentIconMock from '../../images/ParentIcon-Mock.jpg'
import ListPage from '../../components/List'
import parentStore from '../../stores/ParentStore'

import Header from '../../components/Header'

const StyledView = styled.View`
    flex-grow: 1;
    justify-content: center;
    align-items: stretch;
    height: ${props => props.height};
`

const MainView = styled.ImageBackground`
    height: ${props => props.height};
    align-items: center;
`

const styles = StyleSheet.create({
    icon: {
        width: 24,
        height: 24,
    },
});

class LogoTitle extends React.Component {
    render() {
        return (<View>
            <Image
                source={parentIconMock}
                style={{ width: 40, height: 40 }}
            />
            <Text style={styles.message}>
                Hello {parentStore.parentCalled}!
            </Text>
        </View>);
    }
}

type Props = {
    navigator: {}
};
@observer
export default class Home extends Component<Props> {

    static navigationOptions = {
        headerTitle: <Header welcomeMessage={`Welcome ${parentStore.parentCalled}`}  />,
    };

    render() {
        const { navigation } = this.props;
        const { childrenStore } = navigation.state.params;
        return (
            <StyledView height={Dimensions.get('window').height}>
                <MainView
                    height={Dimensions.get('window').height}
                    source={backgroundBlue}
                >
                    <ListPage
                        navigation={navigation}
                        childrenStore={childrenStore}
                    />
                </MainView>

            </StyledView>
        );
    }
}