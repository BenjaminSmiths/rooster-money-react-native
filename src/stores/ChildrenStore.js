import { observable, computed, reaction, action } from 'mobx';
import { API_URL, ACCESS_KEY, ACCESS_PASSWORD } from 'react-native-dotenv'

import parentStore, { ParentStore } from './ParentStore'

import BalanceModel from '../models/BalanceModel'

export default class ChildrenStore {
    @observable balances = [];
    @observable goals = [];

    parentStore: ParentStore

    getBalance() {
        reaction(
            () => this.parentStore.isAuthenticated,
            () => {
                return fetch(`${API_URL}/v1/balance`, {
                    method: 'get',
                    headers: new Headers({
                        'Content-Type': 'application/json',
                        'x-access-token': this.parentStore.getToken
                    })
                })
                    .then(response => response.json())
                    .then(balances => {
                        this.balances = balances.map(balance => new BalanceModel(this, balance))
                    })
            }
        );
    }

    static create(parentStore) {
        const childrenStore = new ChildrenStore();
        childrenStore.parentStore = parentStore
        return childrenStore;
    }
}