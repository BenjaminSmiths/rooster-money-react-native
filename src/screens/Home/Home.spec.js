import React from 'react';
import { Provider } from 'mobx-react';
import renderer from 'react-test-renderer';
import ChildrenStore from '../../stores/ChildrenStore'
import parentStore from '../../stores/ParentStore';
import HomeScreen from './HomeScreen';


test('Home Screen Renders Correctly', () => {
    // Given
    const childrenStore = ChildrenStore.create(parentStore)
    const mockNavigation = { state: { params: { childrenStore } } };

    // When
    const tree = renderer.create(
        <Provider>
            <HomeScreen navigation={mockNavigation}/>
        </Provider>
    ).toJSON();

    // Then
    expect(tree).toMatchSnapshot();
});