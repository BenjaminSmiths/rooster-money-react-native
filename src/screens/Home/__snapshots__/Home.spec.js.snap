// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Renders Correctly 1`] = `
<View
  height={1334}
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "flexGrow": 1,
        "height": 1334,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 1334,
        },
      ]
    }
  >
    <Image
      height={1334}
      source={
        Object {
          "testUri": "../../../src/images/background-blue.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Array [
            Object {
              "paddingTop": 50,
            },
          ]
        }
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  height={1334}
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "flexGrow": 1,
        "height": 1334,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 1334,
        },
      ]
    }
  >
    <Image
      height={1334}
      source={
        Object {
          "testUri": "../../../src/images/background-blue.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Array [
            Object {
              "paddingTop": 50,
            },
          ]
        }
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</View>
`;
